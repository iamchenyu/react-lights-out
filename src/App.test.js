import React from "react";
import { render } from "@testing-library/react";
import App from "./App";

beforeEach(() => {
  jest.spyOn(global.Math, "random").mockReturnValue(0.5);
});

afterEach(() => {
  jest.spyOn(global.Math, "random").mockRestore();
});

it("renders without crashing", () => {
  render(<App />);
});

it("matches the snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});
