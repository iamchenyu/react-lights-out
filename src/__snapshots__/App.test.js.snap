// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <h1>
      Lights out
    </h1>
    <table
      class="Board"
    >
      <tbody>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
        </tr>
        <tr>
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
          <td
            class="Cell Cell-lit"
            data-testid="cell"
            role="button"
          />
        </tr>
      </tbody>
    </table>
  </div>
</DocumentFragment>
`;
